### PR TITLE
Add settings API

### PR DESCRIPTION
**Pull request by soupette**
### What does it do?

Describe the technical changes you did.

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/admin/admin/src/StrapiApp.js [hunk 1]
```diff
@@ -73,13 +84,41 @@ class StrapiApp {
...
+      `link.Component should be a valid React Component`
+    );
+
+    this.settings[sectionId].links.push(link);
+  };
+
+  addSettingsLinks = (sectionId, links) => {
+    invariant(this.settings[sectionId], 'The section does not exist');
+
+    links.forEach(link => {
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/StrapiApp.js:109]: Maybe we can check that links is a valid array here?

File: packages/core/admin/admin/src/StrapiApp.js [hunk 2]
```diff
@@ -90,11 +129,31 @@ class StrapiApp {
...
+        boot({
+          addSettingsLink: this.addSettingsLink,
+          addSettingsLinks: this.addSettingsLinks,
+          getPlugin: this.getPlugin,
+        });
       }
     });
   }
 
+  createSection = (section, links) => {
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/StrapiApp.js:141]: Can we be more specific about the name of the function to clearly understand what it does? Something like `createSettingSection` might be more relevant, WDYT? Without knowing the API, I don't know what section will be created.

File: packages/core/admin/admin/src/StrapiApp.js [hunk 3]
```diff
@@ -90,11 +129,31 @@ class StrapiApp {
...
+    invariant(
+      section.intlLabel?.id && section.intlLabel?.defaultMessage,
+      'section.intlLabel should be defined'
+    );
+    invariant(links, 'links should be defined');
+    invariant(this.settings[section.id] === undefined, 'A similar section already exists');
+
+    this.settings[section.id] = { ...section, links: [] };
+
+    links.forEach(link => {
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/StrapiApp.js:152]: Same here, maybe we can check if `links` is an array

File: packages/core/admin/admin/src/hooks/useSettingsMenu/utils/sortLinks.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import sortBy from 'lodash/sortBy';
+
+const sortLinks = links => sortBy(links, object => object.id);
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/hooks/useSettingsMenu/utils/sortLinks.js:3]: Can you rename `object` to `link` please?

File: packages/core/admin/admin/src/pages/SettingsPage/tests/index.test.js [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+    userEvent.click(screen.getByText('i18n.plugin.name'));
+
+    expect(screen.getByText(/i18n settings/)).toBeInTheDocument();
+
+    userEvent.click(screen.getByText('email'));
+
+    expect(screen.getByText(/email settings/)).toBeInTheDocument();
+  });
+});
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/pages/SettingsPage/tests/index.test.js:338]: Nice tests!

File: packages/core/admin/admin/src/pages/SettingsPage/utils/defaultRoutes.js [hunk 6]
```diff
@@ -9,11 +8,6 @@ import WebhooksEditView from '../../Webhooks/ProtectedEditView';
 import WebhooksListView from '../../Webhooks/ProtectedListView';
 
 const defaultRoutes = [
-  {
-    Component: ApplicationInfosPage,
-    to: '/settings/application-infos',
-    exact: true,
-  },
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/pages/SettingsPage/utils/defaultRoutes.js:16]: We don't need that link?

**Review by HichamELBSI — CHANGES_REQUESTED** [review]
Good job @soupette!!

File: packages/core/admin/admin/src/pages/SettingsPage/utils/defaultRoutes.js (hunk 6 above)
Comment (soupette) [reply packages/core/admin/admin/src/pages/SettingsPage/utils/defaultRoutes.js:16]: I have created a dedicated Route for this since it is a static one.

**Review by HichamELBSI — APPROVED** [review]
LGTM great job!!!

**Pull request merged**